src/Data/Vec/Instances.agda

```diff
@@ -9,7 +9,16 @@
 module Data.Vec.Instances where
 
 open import Data.Vec.Categorical
+open import Data.Vec.Properties
+  using (≡-dec)
+open import Relation.Binary.PropositionalEquality.Properties
+  using (isDecEquivalence)
+open import Data.Vec.Relation.Binary.Equality.DecPropositional
+open import Relation.Binary.TypeClasses
+
 
 instance
   vecFunctor = functor
   vecApplicative = applicative
+
+  ≡-isDecEquivalence-Vec = λ {a} {A} {{≡-isDecEquivalence-A}} {n} → isDecEquivalence (≡-dec {a} {A} (_≟_ {{≡-isDecEquivalence-A}}) {n})
```

Comment:
Hmm we should also try and make the naming conventions match. So far we have `vecFunctor` and `vecApplicative` but here we have `≡-isDecEquivalence-Vec`. Would people object to `vecIsDecEquivalence`? It's not like we're going to have non-propositional equality instances right? Or should we be going for some other convention?

Reply:
`vecIsDecEquivalence` is strange though because it reads as if `Vec` itself would be a decidable equivalence, which is nonsense. In fact it does not even mention the name of the relation that is decidable! However, I could go for something like `vec≡-isDecidableEquivalence`, which can be read naturally as "Vector equality is a decidable equivalence".

Reply:
Hmm that's a good point. `vec≡-isDecidableEquivalence` is better, but we usually use `-` to separate components which would make it `vec-≡-isDecidableEquivalence`. But I also like your capitalisation of the types in your original proposal as well. Hmm....

I'm aware we've almost gone full circle, but how about `Vec-≡-isDecidableEquivalence`, and I later go back and deprecate `vecFunctor` etc. in favour of `Vec-functor`?

Reply:
Why do we need the `vec` prefix in `Data.Vec.Instances`?

Reply:
In a typical use case of these type classes, you'd want to open multiple of these `Instances` modules at the same time (otherwise, what's the point?) so the names should not cause conflicts.

Reply:
But that's what qualified names are for?

Reply:
I find it difficult to really care either way since the whole point of instance search is that you're not going to give these names explicitly. However, one argument in favour of the unambiguous names is that they can be used together with the `--no-qualified-instances` flag, which can help with controlling the scope of instances (see https://github.com/agda/agda/pull/4522).

Reply:
The ambiguous names should have no problems with `--no-qualified-instances` either. It's perfectly fine to say
```agda
open import Data.Vec.Instances using (functor)
open import Data.List.Instances using (functor)
```
as long as you don't write `functor` yourself. Instance search will pick up the correct unqualified instance.